ledger-tool/src/main.rs

```diff
@@ -1112,6 +1120,76 @@ fn main() {
+                dev_halt_at_slot,
+                new_hard_forks: hardforks_of(arg_matches, "hard_forks"),
+                poh_verify: false,
+                ..ProcessOptions::default()
+            };
+            let genesis_config = open_genesis_config_by(&ledger_path, arg_matches);
+            match load_bank_forks(arg_matches, &ledger_path, &genesis_config, process_options) {
+                Ok((bank_forks, _leader_schedule_cache, _snapshot_hash)) => {
+                    let slot = bank_forks.working_bank().slot();
+                    let bank = bank_forks.get(slot).unwrap_or_else(|| {
+                        eprintln!("Error: Slot {} is not available", slot);
+                        exit(1);
+                    });
+
+                    use solana_sdk::native_token::LAMPORTS_PER_SOL;
+                    use std::fmt::{Display, Formatter, Result};
+                    pub struct Sol(u64);
+
+                    impl Display for Sol {
+                        fn fmt(&self, f: &mut Formatter) -> Result {
+                            write!(
+                                f,
+                                "{}.{:09} SOL",
+                                self.0 / LAMPORTS_PER_SOL,
+                                self.0 % LAMPORTS_PER_SOL
+                            )
+                        }
+                    }
+
+                    let computed_capitalization: u64 = bank
+                        .get_program_accounts(None)
+                        .into_iter()
+                        .filter_map(|(_pubkey, account)| {
+                            if account.lamports != u64::max_value() {
+                                let is_specially_retained =
+                                    solana_sdk::native_loader::check_id(&account.owner)
+                                        || solana_sdk::sysvar::check_id(&account.owner);
+
+                                if is_specially_retained {
+                                    // specially retained accounts are ensured to exist by
+                                    // alwaysing having a balance of 1 lamports, which is
+                                    // outside the capitalization calculation.
+                                    Some(account.lamports - 1)
+                                } else {
+                                    Some(account.lamports)
+                                }
+                            } else {
+                                None
+                            }
+                        })
+                        .sum();
+
+                    if bank.capitalization() != computed_capitalization {
+                        panic!(
+                            "Capitalization mismatch!?: {} != {}",
+                            bank.capitalization(),
+                            computed_capitalization
+                        );
+                    }
+                    println!("Capitalization: {}", Sol(bank.capitalization()));
```

Comment:
@danpaul000 FYI: I've created the rust version of `capitalization` bash script, which you created in that past iirc. :)

```
$ RUST_LOG=solana=error solana-ledger-tool --ledger config/ledger/ capitalization
Capitalization: 999999999.998820000 SOL
```